### ISSUE TITLE
CLI Formatter

### ISSUE THREAD

**Issue by neenjaw** [exercism/elixir-test-runner#1]
For the ExUnit component, we can specify the CLI Formatter from the command line, so if we reimplement the default [CLIFormatter module](https://github.com/elixir-lang/elixir/blob/v1.9.1/lib/ex_unit/lib/ex_unit/cli_formatter.ex)'s `handle_cast({:suite_finished, run_us, load_us}, config)` ([line](https://github.com/elixir-lang/elixir/blob/79388035f5391f0a283a48fba792ae3b4f4b5f21/lib/ex_unit/lib/ex_unit/cli_formatter.ex#L34-36)) to output the results map structure in the `config` variable as a json, we can just print it to the output, and capture it to a file.

We would need to insert this module and also maybe the `Jason` JSON interpreter from hex into the mix project, then can be run as easily as `mix test --formatter "ExercismTest.CLIFormatter"` I think.

This way we don't have to parse the existing output strings.

**Comment by neenjaw** [comment]
I would be interested in reimplementing the formatter if this is the route we want to take.

**Comment by NobbZ** [comment]
I'm not a fan of putting unnneeded dependencies into the mix.exs. It should be possible though to "burn" them into the docker image using `mix archive` or by mimicking its behaviour.

Alternatively creating a copy of the sources and artifacts ind `_build` and `deps` respectively should work.

**Comment by neenjaw** [comment]
I agree with you about it being a little messy to mess around with the dependencies, I don't think I know how to use `mix archive` but from what you say, it sounds like the better way to include what's needed.

**Comment by NobbZ** [comment]
Basically its putting the compiled modules in `~/.mix/archives/$name/$name/ebin`. `mix` will always have `~/.mix/archives/*/*/ebin` in the module lookup path for the BEAM.

**Comment by neenjaw** [comment]
Interesting!

**Comment by neenjaw** [comment]
Ok, so I have the formatter pending merge in PR and I am working on another branch which will add the run shell script for the docker entrypoint

**Comment by neenjaw** [comment]
Ok... so I am trying to test this out, but I am having trouble testing the implementation you described, @NobbZ.

These are the steps I am taking:
1. Compiling the formatter mix project
1. `cd` to the submission folder
1. `mix compile` the submission
1. recursively copying (`cp -r`) the `_build/dev/lib` from the json formatter project to a submission project
1. testing the submission `mix test --no-compile --include pending:true --formatter JSONFormatter`

To which I got no response.  So I went into `iex -S mix`

```shell
iex(7)> Code.ensure_compiled?(ExUnit.CLIFormatter) 
true
iex(8)> Code.ensure_compiled?(JSONFormatter)       
false
```

So it seems like it can't find the compiled module.

Even manually specifying in the test_helper.exs I can't get it to be found:

```elixir
ExUnit.configure(formatters: [JSONFormatter, ExUnit.CLIFormatter])
ExUnit.start()
```

Thoughts about what I'm missing?
